src/hooks/useConsoleDebugger.ts

```diff
@@ -0,0 +1,63 @@
+import getConfig from 'next/config';
+import { useCallback, useMemo } from 'react';
+
+import { FeatureFlags, useFeatureFlag } from '@/hooks/useFeatureFlag';
+
+const useConsoleDebugger = () => {
+  /* eslint-disable no-console */
+  const [showDebugging] = useFeatureFlag(FeatureFlags.SHOW_CONSOLE_DEBUGGING);
+
+  const isDevelopment = useMemo(() => {
+    const { publicRuntimeConfig } = getConfig() || {};
+    return publicRuntimeConfig?.environment === 'development';
+  }, []);
```

Comment:
useMemo is not necessary here.

```suggestion
const { publicRuntimeConfig } = getConfig() || {};
const isDevelopment = publicRuntimeConfig?.environment === 'development';
```